plugins/generator-1.20.1/forge-1.20.1/procedures/team_get_friendlyfire.java.ftl

```diff
@@ -0,0 +1,9 @@
+(new Object() {

```

Comment:
Those could probably be reduced to inline methods with ternary operator, applies to friendly fire and get entity team.

This way we avoid Object creation which is our last resort only when really needed

Reply:
Eg. entity_has_item_inventory

Reply:
This makes the code ugly, messy and just bad.
```java
((world instanceof Level _level) && _level.getScoreboard().getPlayerTeam(${input$name}) != null && _level.getScoreboard().getPlayerTeam(${input$name}).isAllowFriendlyFire())
```
or if you want to use `?` and `:`, it's even worse
 ```java
((world instanceof Level _level) ? _level.getScoreboard().getPlayerTeam(${input$name}) != null ? _level.getScoreboard().getPlayerTeam(${input$name}).isAllowFriendlyFire() : false : false)
```

Reply:
We already use that approach with some blocks AFAIK. About ternary operator, it is mostly needed when working with data other than boolean so not really needed in your case (on a side note, you could use `a & b ? c : false` rather than changing `a & b & c` directly to `a ? b ? c : false : false`).

Reply:
The performance burden of "new Object" allocating on heap and fragmentation heap will not be accepted.

You will need to change the way we use in other locations. Check existing code for inspiration